src/rgw/rgw_rados.cc

```diff
@@ -11434,6 +11460,61 @@ int RGWRados::bucket_index_read_olh_log(const RGWBucketInfo& bucket_info, RGWObj
   return 0;
 }
 
+// a multisite sync bug resulted in the OLH head attributes being overwritten by
+// the attributes from another zone, causing link_olh() to fail endlessly due to
+// olh_tag mismatch. this attempts to detect this case and reconstruct the OLH
+// attributes from the bucket index. see http://tracker.ceph.com/issues/37792
+int RGWRados::repair_olh(RGWObjState* state, const RGWBucketInfo& bucket_info,
+                         const rgw_obj& obj)
+{
+  // fetch the current olh entry from the bucket index
+  rgw_bucket_olh_entry olh;
+  int r = bi_get_olh(bucket_info, obj, &olh);
+  if (r < 0) {
+    ldout(cct, 0) << "repair_olh failed to read olh entry for " << obj << dendl;
+    return r;
+  }
+  if (olh.tag == rgw_bl_str(state->olh_tag)) { // mismatch already resolved?
```

Comment:
@cbodley The FTBFS seems to be emanating from this line - namely, `rgw_bl_str` is not defined. 

To me, that seems to indicate I need to either (a) cherry-pick dcb5e06ca1d or (b) replace `rgw_bl_str(state->olh_tag)` with `state->olh_tag.to_str()`

I'm leaning towards (b)...

Reply:
. . . taking a stab at cherry-picking dcb5e06 - just for kicks, as an experiment

Reply:
. . . and I now see that you went with `state->olh_tag.to_str()` in the luminous backport https://github.com/ceph/ceph/pull/26191